ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -5,78 +5,130 @@
-import type { TtlEvent } from 'vault/app-types';
+import type LeaseConfigModel from 'vault/models/aws/lease-config';
+import type RootConfigModel from 'vault/models/aws/root-config';
+import type Router from '@ember/routing/router';
+import type Store from '@ember-data/store';
+import type FlashMessageService from 'vault/services/flash-messages';
 
 /**
- * @module ConfigureAwsComponent
+ * @module ConfigureAwsComponent is used to configure the AWS secret engine
+ * A user can configure the endpoint root/config and/or lease/config. 
+ * The fields for these endpoints are on one form.
  *
  * @example
  * ```js
  * <SecretEngine::ConfigureAws
-    @model={{model}}
-    @tab={{tab}}
-    @accessKey={{accessKey}}
-    @secretKey={{secretKey}}
-    @region={{region}}
-    @iamEndpoint={{iamEndpoint}}
-    @stsEndpoint={{stsEndpoint}}
-    @saveAWSRoot={{action "save" "saveAWSRoot"}}
-    @saveAWSLease={{action "save" "saveAWSLease"}} />
+    @rootConfig={{this.model.aws-root-config}}
+    @leaseConfig={{this.model.aws-lease-config}} 
+    @id={{this.model.id}} 
+    />
  * ```
  *
- * @param {object} model - aws secret engine model
- * @param {string} tab - current tab selection
- * @param {string} accessKey - AWS access key
- * @param {string} secretKey - AWS secret key
- * @param {string} region - AWS region
- * @param {string} iamEndpoint - IAM endpoint
- * @param {string} stsEndpoint - Sts endpoint
- * @param {Function} saveAWSRoot - parent action which saves AWS root credentials
- * @param {Function} saveAWSLease - parent action which updates AWS lease information
- *
+ * @param {object} rootConfig - AWS config/root model
+ * @param {object} leaseConfig - AWS config/lease model
+ * @param {string} id - name of the AWS secret engine, ex: 'aws-123'
  */
 
-type AWSRootCredsFields = {
-  access_key: string | null;
-  iam_endpoint: string | null;
-  sts_endpoint: string | null;
-  secret_key: string | null;
-  region: string | null;
-};
-
-type LeaseFields = { lease: string; lease_max: string };
-
 interface Args {
-  model: SecretEngineModel;
-  tab?: string;
-  accessKey: string;
-  secretKey: string;
-  region: string;
-  iamEndpoint: string;
-  stsEndpoint: string;
-  saveAWSRoot: (data: AWSRootCredsFields) => void;
-  saveAWSLease: (data: LeaseFields) => void;
+  leaseConfig: LeaseConfigModel;
+  rootConfig: RootConfigModel;
+  id: string;
 }
 
 export default class ConfigureAwsComponent extends Component<Args> {
-  @action
-  saveRootCreds(data: AWSRootCredsFields, event: Event) {
+  @service declare readonly router: Router;
+  @service declare readonly store: Store;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked errorMessageRoot: string | null = null;
+  @tracked errorMessageLease: string | null = null;
+  @tracked invalidFormAlert: string | null = null;
+  @tracked modelValidationsLease: ValidationMap | null = null;
+
+  @task
+  @waitFor
+  *save(event: Event) {
```

Comment:
This method is long, and while the comments are helpful I think it might be even more helpful if the we had the saveRoot action and saveLease action in separate functions